### PR TITLE
Checkout: Move LineItem and NonProductLineItem to wpcom-checkout package

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This PR adjusts the `LineItem` and `NonProductLineItem` components used in checkout so that they have no Calypso (or Redux) dependencies. It then moves these components to the `@automattic/wpcom-checkout` package. 

By moving the components to a package outside Calypso, it becomes possible to use them in non-Calypso contexts like wp-admin. It also guards against regressions because even though the components are already TypeScript, TypeScript errors are ignored by Calypso while the `wpcom-calypso` package will not compile if there are any present.

This is part of https://github.com/Automattic/wp-calypso/issues/52215

This is needed by https://github.com/Automattic/wp-calypso/pull/52890

Requires D67072-code and D67068-code to be merged in order for P2 Plus and Jetpack Search plan subtitles to be displayed correctly but this should be testable without them.

Here's a screenshot of a Jetpack Search plan subtitle before and after this change:

jetpack-search-checkout-subtitle

Here's a screenshot of a p2 plan subtitle before this change:

p2-checkout-subtitle-before

And here's how it looks after this change (a slight formatting difference in how the price is displayed):

p2-checkout-subtitle

#### Testing instructions

- Add a product to the cart and visit checkout.
- Verify that the line items are displayed with no errors.

### CODE REVIEW

File: packages/wpcom-checkout/src/checkout-line-items.tsx
Comment (nbloomf) [inline packages/wpcom-checkout/src/checkout-line-items.tsx:164]: There's no reason to do anything about it now, but `isPwpoUser` seems business-logic specific enough that I wonder if it would be better to pass around a user flags object instead. With only one flag that would be overkill. But reading this my first thought was "why does the line item component need to know this?", and the answer is that it doesn't, except that the modal copy depends on it. That is, the specificity of `isPwpoUser` is kind of a red herring.

**Review by nbloomf — APPROVED** [review]
Looks good and works as described!

**Comment by sirbrillig** [comment]
Rebased. Just waiting on the backend changes to merge.

**Comment by sirbrillig** [comment]
Adjusted the new property names after the backend diffs merged since they modified the names slightly. I re-tested the affected product types and put updated screenshots of the results in the PR description. Everything seems good.

**Pull request merged**
